app/api/sul_bib/authorship_api.rb

```diff
@@ -113,11 +114,8 @@ def get_publication_via_sciencewire!(sw_id)
       # @param [String] wos_uid WebOfScience ID
       # @return [Publication]
       def get_publication_via_wos!(author, wos_uid)
-        pub = Publication.find_by(wos_uid: wos_uid)
-        return pub if pub
-        uids = WebOfScience.harvester.process_uids(author, [wos_uid])
-        uids.present? || log_and_error!("The #{wos_uid} publication was not found either locally or at WebOfScience.")
-        Publication.find_by(wos_uid: uids.first)
+        WebOfScience.harvester.author_uid(author, wos_uid) ||
```

Comment:
so if I am reading this correctly, the updated process records class below will be called by harvester and take care of the discovery of existing WoS source records at publication creation time, right?

Reply:
`author_uid` calls `author_contributions` and/or `process_records`.  The former queries `Publication` by `wos_uid` (and backfills contribs if needed).  `process_records` handles the case where nothing matches, or only a source record matches.